### PR TITLE
AKCORE-15: Acknowledge API for Share Partition (1/N)

### PR DESCRIPTION
**Pull request by apoorvmittal10**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/java/kafka/server/SharePartition.java [hunk 1]
```diff
@@ -32,208 +38,256 @@
...
     }
 
-    public void setMessageState(RecordState newState, boolean incrementDeliveryCount) {
-      state = newState;
-      if (incrementDeliveryCount) {
-        deliveryCount++;
+  static class InFlightFetchBatch {
+
+      // The member id of the client that is fetching the record.
+      private final String memberId;
```
Comment (AndrewJSchofield) [inline core/src/main/java/kafka/server/SharePartition.java:186]: I was expecting Uuid. If we are generally using String, that's OK I guess, but a Uuid should be slightly more efficient for comparison and storage.

File: core/src/main/java/kafka/server/SharePartition.java [hunk 2]
```diff
@@ -32,208 +38,256 @@
...
+          }
+      }
+
+      @Override
+      public String toString() {
+          return "InFlightFetchBatch(" +
+              " memberId=" + memberId +
+              ", startOffset=" + startOffset +
+              ", lastOffset=" + lastOffset +
+              " state=" + state.toString() +
```
Comment (AndrewJSchofield) [inline core/src/main/java/kafka/server/SharePartition.java:262]: Missing "," ?

File: core/src/main/java/kafka/server/SharePartition.java [hunk 3]
```diff
@@ -32,208 +38,256 @@
...
+      }
   }
 
   static class AcknowledgementBatch {
 
-    private final long startOffset;
-    private final long lastOffset;
-    private final List<Long> gapOffsets;
-    private final byte acknowledgeType;
+      private final long startOffset;
```
Comment (AndrewJSchofield) [inline core/src/main/java/kafka/server/SharePartition.java:271]: This is a mistake in the request schema. Please use `baseOffset` for this variable. I'll update the schema next time we do a refresh.

**Review by AndrewJSchofield — APPROVED** [review]

File: core/src/main/java/kafka/server/SharePartition.java (hunk 1 above)
Comment (apoorvmittal10) [reply core/src/main/java/kafka/server/SharePartition.java:186]: Agree, just went for now with the memberid as defined in the request. Shall we change the fetch request as well?

File: core/src/main/java/kafka/server/SharePartition.java (hunk 2 above)
Comment (apoorvmittal10) [reply core/src/main/java/kafka/server/SharePartition.java:262]: Done.

File: core/src/main/java/kafka/server/SharePartition.java (hunk 3 above)
Comment (apoorvmittal10) [reply core/src/main/java/kafka/server/SharePartition.java]: Done.

File: core/src/main/java/kafka/server/SharePartition.java (hunk 1 above)
Comment (apoorvmittal10) [reply core/src/main/java/kafka/server/SharePartition.java:186]: I am merging for now, I think we should change the request schema as well.

**Pull request merged**